From b61f428dbd31b04fc7d5d8d6396cf375d2a17a5d Mon Sep 17 00:00:00 2001
From: Mike Gerdts <mike.gerdts@joyent.com>
Date: Thu, 29 Nov 2018 22:22:26 +0000
Subject: [PATCH] OS-7411 manifest.d/live.manifest collects temporary files
 Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by: Dan McDonald
 <danmcd@joyent.com> Approved by: Dan McDonald <danmcd@joyent.com>

---
 src/Makefile | 33 ++++++++++++++++-----------------
 1 file changed, 16 insertions(+), 17 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index def0d787..852e7bf8 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -487,23 +487,22 @@ check: $(JSLINT) $(SUBDIRS)
 
 .PHONY: manifest
 manifest:
-	(cat manifest && cd vm && \
-	    echo "# list of tests is autogenerated by smartos-live/src/Makefile" && \
-	    (find tests/ -type f -exec echo "{}" "0444 root bin" \; | grep -v "/testdata/")) \
-	    | sed -e "s|^tests/|f usr/vm/test/tests/|" \
-	    > $(DESTDIR)/$(DESTNAME)
-	(cd vm/tests && \
-	    find testdata/ -type f -exec echo "{}" "0444 root bin" \;) \
-	    | sed -e "s|^testdata/|f usr/vm/test/testdata/|" \
-	    >> $(DESTDIR)/$(DESTNAME)
-	(cd fw/test && \
-	    find integration/ -type f -exec echo "{}" "0444 root bin" \;) \
-	    | sed -e "s|^integration/|f usr/fw/test/integration/|" \
-	    >> $(DESTDIR)/$(DESTNAME)
-	(cd fw/etc && \
-	    find examples/ -type f -exec echo "{}" "0444 root bin" \;) \
-	    | sed -e "s|^examples/|f usr/fw/etc/examples/|" \
-	    >> $(DESTDIR)/$(DESTNAME)
+	cp manifest ./$(DESTNAME).tmp
+	echo "# list of tests is autogenerated by smartos-live/src/Makefile" \
+	    >> ./$(DESTNAME).tmp
+	(cd vm && git ls-files tests) | grep -v /testdata/ \
+	    | sed -e 's|^|f usr/vm/test/|' -e 's|$$| 0444 root bin|' \
+	    >> ./$(DESTNAME).tmp
+	(cd vm/tests && git ls-files testdata) \
+	    | sed -e 's|^|f usr/vm/test/|' -e 's|$$| 0444 root bin|' \
+	    >> ./$(DESTNAME).tmp
+	(cd fw/test && git ls-files integration) \
+	    | sed -e 's|^|f usr/fw/test/|' -e 's|$$| 0444 root bin|' \
+	    >> ./$(DESTNAME).tmp
+	(cd fw/etc && git ls-files examples) \
+	    | sed -e 's|^|f usr/fw/etc/|' -e 's|$$| 0444 root bin|' \
+	    >> ./$(DESTNAME).tmp
+	mv ./$(DESTNAME).tmp $(DESTDIR)/$(DESTNAME)
 
 .PHONY: mancheck_conf
 mancheck_conf:
-- 
2.21.0

